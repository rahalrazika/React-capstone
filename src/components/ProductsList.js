import React, { useEffect, useState } from 'react';
import { connect, useDispatch } from 'react-redux';
import axios from 'axios';
import Product from './Product';
import { listProducts } from '../actions/productActions';
import Loading from './loading';
import banner from '../assets/banner.jpg';

const ProductsList = () => {
  const [loading, setLoading] = useState(false);

  const dispatch = useDispatch();
  const fetchProducts = async () => {
    const response = await axios
      .get('https://fakestoreapi.com/products')
      .catch((err) => err);
    dispatch(listProducts(response.data));
  };

  useEffect(() => {
    fetchProducts();
    setLoading(true);
  }, []);
  return (
    <div className="productContainer">
      <div className="banner">
        <img src={banner} alt="banner" />
      </div>
      <div className="container d-flex">
        {loading ? <Product /> : <Loading />}
      </div>
    </div>
  );
};

export default connect(null, { listProducts })(ProductsList);
