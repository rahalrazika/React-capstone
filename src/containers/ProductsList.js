import React, { useEffect, useState } from 'react';
import { connect, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import axios from 'axios';
import Product from './Product';
import { listProducts, filterProducts } from '../actions/productActions';
import Loading from '../components/loading';
import banner from '../assets/banner.jpg';

const ProductsList = (props) => {
  const { filteredProducts, filterProducts } = props;
  const [loading, setLoading] = useState(false);

  const dispatch = useDispatch();
  const fetchProducts = async () => {
    const response = await axios
      .get('https://fakestoreapi.com/products')
      .catch((err) => err);
    dispatch(listProducts(response.data));
    filterProducts(response.data);
  };

  useEffect(() => {
    fetchProducts();
    setLoading(true);
  }, []);
  return (
    <div className="productContainer">
      <div className="banner">
        <img src={banner} alt="banner" />
      </div>
      <div className="container d-flex">
        {loading ? (
          filteredProducts.map((singleProduct) => (
            <Product key={singleProduct.id} productData={singleProduct} />
          ))
        ) : (
          <Loading />
        )}
      </div>
    </div>
  );
};
ProductsList.propTypes = {
  filteredProducts: PropTypes.instanceOf(Array),
  filterProducts: PropTypes.func,

};

ProductsList.defaultProps = {
  filteredProducts: [],
  filterProducts: null,
};

const mapStateToProps = (state) => ({
  filteredProducts: state.listOfProduct.filteredProducts,
});

export default connect(mapStateToProps, { listProducts, filterProducts })(
  ProductsList,
);
