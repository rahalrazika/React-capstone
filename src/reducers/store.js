import { createStore } from 'redux';
import rootReducer from './index';

const store = createStore(rootReducer);
export default store;
