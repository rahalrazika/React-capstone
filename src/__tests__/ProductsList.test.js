import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ProductsList from '../containers/ProductsList';
import store from '../reducers/store';

it('expect to render productList components', () => {
  const result = renderer.create(
    <Provider store={store}>
      <ProductsList />
    </Provider>,

  ).toJSON;
  expect(result).toMatchSnapshot();
});
