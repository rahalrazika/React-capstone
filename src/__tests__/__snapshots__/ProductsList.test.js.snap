// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render productList components 1`] = `[Function]`;
